socorro/processor/rules/breakpad.py

```diff
@@ -146,30 +144,6 @@ def action(self, raw_crash, dumps, processed_crash, status):
                     self.truncate_stacks(value["json_dump"], f"{key}.json_dump", status)
 
 
-@contextmanager
-def tmp_raw_crash_file(tmp_path, raw_crash, crash_id):
-    """Saves JSON data to file, returns path, and deletes file when done.
-
-    :param tmp_path: str path to temp storage
-    :param raw_crash: dict of raw crash data
-    :param crash_id: crash id for this crash report
-
-    :yields: absolute path to temp file
-
-    """
-
-    path = os.path.join(
-        tmp_path, f"{crash_id}.{threading.currentThread().getName()}.TEMPORARY.json"
```

Comment:
This uses the thread name to prevent contention between threads processing the same crash report. However, when we switch from multi-threaded to multi-process, this wouldn't work anymore. This was another reason to redo how the processor manages temporary files/directories during processing.